compositor_common/src/scene/node.rs

```diff
@@ -30,10 +35,35 @@ pub enum NodeParams {
 }
 
 impl NodeSpec {
-    pub fn validate(&self) -> Result<(), NodeSpecValidationError> {
+    pub fn validate_params(&self) -> Result<(), NodeSpecValidationError> {
         match &self.params {
-            NodeParams::Builtin { transformation, .. } => Ok(transformation.validate(self)?),
+            NodeParams::Builtin { transformation } => Ok(transformation.validate_params(self)?),
             _ => Ok(()),
         }
     }
+
+    pub fn identification_name(&self) -> Rc<str> {
```

Comment:
Also, I would avoid Rc<str> there, any time you call that function that string will be immediately used to create an error, so it's never copied. Also because of that you are making errors that don't use it not Send.

I think that w either should use:
- Arc<str> for stuff like ids that we are passing around
- Arc<str> for absolutely any string that is not &'static str

Anything other than that will be confusing when you are reading the code.